### PR TITLE
[Feature 6.1] & [Feature 6] Updated Use Case and Screens

### PR DESCRIPTION
**Pull request by helenaglow**
This is my screens for now and the updated use case (not sure why there are so many commits github scares me)

- I have uploaded @hannahpan1 UserPublicProfile as it was not merged at the time and it necessary for my code to work.

- Also, the screens are not connected to the controller rn as this is just to test.

- Implemented Strategy design for the Matching Algorithm as TA requested.

- I have addded my version of main just so you can see how I got it to work

### Please let me know if I have code smells or violated clean architecture! Thanks

TO-DO:
- [ ] Connect to Aarya's presenting screen
- [X] Connect to Ipek's use case
- [X] Improve Javadocs
- [X] Implement testing (I am having a problem with my tests rn "Test not found" and I am using the jupiter)

### CODE REVIEW

File: src/main/java/Entities/UserPublicProfile.java [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
     public UserPublicProfile(){
-//        this.preferences = HashMap<String, String> ;
+        HashMap<String, String> preferences = new HashMap<>();
+        /*Prefilling in the preferences*/
+        preferences.put("Time Commitment", "");
+        preferences.put("Location", "");
+        preferences.put("Meeting Time", "");
+        this.preferences = preferences;
+    }
+// HANNAH'S CODE FROM GITHUB SO THERE WILL BE NO CONFLICTS CAUSE WE NEVER MERGED HER STUFF
```
Comment (aaryavatsa) [inline src/main/java/Entities/UserPublicProfile.java:16]: noo this is so upsetting we must merge hannah's asap 💯

**Review by aaryavatsa — DISMISSED** [review]
Everything looks great! Nice work on the Javadocs and the screes -- very clear to see how much effort went into this! Excited to connect my screens to yours. woo

**Review by julia80887 — APPROVED** [review]

**Comment by helenaglow** [comment]
UPDATE: I have updated the home screen and main after aarya's pull request. Also added Tests to the use case!

**Review by aaryavatsa — APPROVED** [review]
looks great!! so excited to link 💯

**Review by soheegoo — APPROVED** [review]

**Pull request merged**